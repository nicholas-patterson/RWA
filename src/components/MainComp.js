import React, { useState } from "react";
import WeatherDetails from "./WeatherDetails";
import axios from "axios";

const MainComp = () => {
  const [zipCode, setZipCode] = useState("");
  const [weatherInfo, setWeatherInfo] = useState(null);
  const [errorMessage, setErrorMessage] = useState("");

  const handleChange = (e) => {
    setZipCode(e.target.value);
  };

  const getWeather = () => {
    if (zipCode == "") {
      setErrorMessage("Field Is Empty");
      setWeatherInfo(null);
    } else {
      setErrorMessage("");
      axios
        .get(
          `http://api.openweathermap.org/data/2.5/weather?zip=${zipCode}&appid=137c7ca49af365102690e25ac22d6569&units=imperial
              `
        )
        .then((res) => {
          setWeatherInfo(res.data);
          setZipCode("");
        })
        .catch((err) => console.log(err));
    }
  };

  return (
    <React.Fragment>
      <form onSubmit={(e) => e.preventDefault()}>
        <input
          data-testid="zipCode-field"
          className="zipCode__field"
          type="text"
          name="zipCode"
          placeholder="Zip Code..."
          onChange={handleChange}
          value={zipCode}
        />
        {errorMessage ? <p className="error">{errorMessage}</p> : ""}
        <div data-testid="button" className="arr" onClick={getWeather}>
          &rarr;
        </div>
      </form>
      {weatherInfo && <WeatherDetails weatherInfo={weatherInfo} />}
    </React.Fragment>
  );
};

export default MainComp;
