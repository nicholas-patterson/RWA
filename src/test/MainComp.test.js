import React from "react";
import { cleanup, render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import MainComp from "../components/MainComp";
import userEvent from "@testing-library/user-event";

afterEach(cleanup);

describe("Main Component", () => {
  it("renders", () => {
    const { asFragment } = render(<MainComp />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders proper text in field", () => {
    const { getByTestId } = render(<MainComp />);

    userEvent.type(getByTestId("zipCode-field"), "19079");
    expect(getByTestId("zipCode-field")).toHaveAttribute("value", "19079");
  });
});
