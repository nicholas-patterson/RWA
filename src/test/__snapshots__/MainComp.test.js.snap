// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders 1`] = `
<DocumentFragment>
  <form>
    <input
      class="zipCode__field"
      data-testid="zipCode-field"
      name="zipCode"
      placeholder="Zip Code..."
      type="text"
      value=""
    />
    <div
      class="arr"
      data-testid="button"
    >
      â†’
    </div>
  </form>
</DocumentFragment>
`;
